sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -132,28 +131,21 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
       REGEXP_EXTRACT_FAMILY, REGEXP_REPLACE)
   }
 
-  private def canFilterLeft(joinType: JoinType): Boolean = joinType match {
-    case Inner | RightOuter => true
-    case _ => false
-  }
-
-  private def canFilterRight(joinType: JoinType): Boolean = joinType match {
-    case Inner | LeftOuter => true
-    case _ => false
-  }
```

Comment:
Move these functions to `JoinSelectionHelper` to make the code easier to maintain.

Reply:
Nice !!!, having this unified it now supports LeftSemi as well, which was added for a follow-up
ref :  https://issues.apache.org/jira/browse/SPARK-38565

Reply:
https://github.com/apache/spark/pull/36131